cpp/benchmarks/io/json/nested_json.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+#include <io/json/nested_json.hpp>
+
+#include <tests/io/fst/common.hpp>
+
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/repeat_strings.hpp>
+#include <cudf/types.hpp>
+
+#include <cstdlib>
+
+namespace cudf {
+namespace {
+auto make_test_json_data(size_type string_size, rmm::cuda_stream_view stream)
+{
+  // Test input
+  std::string input = R"(
+                      {"a":1,"b":2,"c":[3], "d": {}},
+                      {"a":1,"b":4.0,"c":[], "d": {"year":1882,"author": "Bharathi"}},
+                      {"a":1,"b":6.0,"c":[5, 7], "d": null},
+                      {"a":1,"b":null,"c":null},
+                      {
+                        "a" : 1
+                      },
+                      {"a":1,"b":Infinity,"c":[null], "d": {"year":-600,"author": "Kaniyan"}},
+                      {"a": 1, "b": 8.0, "d": { "author": "Jean-Jacques Rousseau"}},)";
+
+  const size_type repeat_times = string_size / input.size();
+
+  auto d_input_scalar   = cudf::make_string_scalar(input, stream);
+  auto& d_string_scalar = static_cast<cudf::string_scalar&>(*d_input_scalar);
+  auto d_scalar         = cudf::strings::repeat_string(d_string_scalar, repeat_times);
+  auto& d_input         = static_cast<cudf::scalar_type_t<std::string>&>(*d_scalar);
+
+  auto generated_json    = std::string(d_input);
+  generated_json.front() = '[';
+  generated_json.back()  = ']';
```

Comment:
This does not look right to me.
The input string is `"(... ,)"` and gets repeated `repeat_times` and so `d_input` would be `"(... ,)(... ,)(... ,) ... (... ,)"` and then `generated_json` would be `"[... ,)(... ,)(... ,) ... (... ,]"`. I don't think this is valid JSON.

Reply:
`(` is part of raw string literal in C++.
(6) Raw string literal in https://en.cppreference.com/w/cpp/language/string_literal
See the example 
```
const char* s1 = R"foo(
Hello
  World
)foo";
```
first character of repeated string is "\n". last character is ",". This code changes them to "[" and "]" respectively.